src/analysis/lattice.h

```diff
@@ -0,0 +1,27 @@
+#ifndef wasm_analysis_lattice_h
+#define wasm_analysis_lattice_h
+
+#include <bitset>
+#include "wasm.h"
+
+namespace wasm::analysis {
+
+enum LatticeComparison {
+    NO_RELATION,
+    EQUAL,
+    LESS,
+    GREATER
+};
+
+template<size_t N>
+struct BitsetPowersetLattice {
+    std::bitset<N> value;
+    static bool isTop(const BitsetPowersetLattice<N> & element);
+    static bool isBottom(const BitsetPowersetLattice<N> & element);
+    static LatticeComparison compare(const BitsetPowersetLattice<N> & left, const BitsetPowersetLattice<N> & right);
+    static BitsetPowersetLattice<N> getLeastUpperBound(const BitsetPowersetLattice<N> & left, const BitsetPowersetLattice<N> & right);
+};
```

Comment:
It would be good for the `BitsetPowersetLattice` to be templated on `T`, the element type of the set, and to have a constructor that takes a `std::vector<T>`. In addition to supplying the set of elements, the vector ensures each element has a unique index for use in the bit vector.

Reply:
I'm leaving this to a follow-up PR.